logsmanagement/parser.c

```diff
@@ -0,0 +1,1622 @@
+            debug(D_LOGS_MANAG, "Item %d (type: REQ_PROC_TIME):%.*s", i, (int)field_size, field);
+            #endif
+
+            if(unlikely(field[0] ==  '-' && field_size == 1)){
+                log_line_parsed->req_proc_time = WEB_LOG_INVALID_PORT;
+                log_line_parsed->parsing_errors++;
+                goto next_item;
+            }
+
+            float f = 0;
+
+            char req_proc_time_d[REQ_PROC_TIME_MAX_LEN];
+            snprintf( req_proc_time_d, 
+                      field_size > (REQ_PROC_TIME_MAX_LEN - 1) ? REQ_PROC_TIME_MAX_LEN : field_size + 1, 
+                      "%.*s", (int) field_size, field);
+
+            if(memchr(field, '.', field_size)){ // nginx time is in seconds with a milliseconds resolution.
+                if(likely(str2float(&f, req_proc_time_d) == STR2XX_SUCCESS)){ 
+                    log_line_parsed->req_proc_time = (int) (f * 1.0E6);
+                }
+                else { 
+                    #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+                    error("Error while extracting REQ_PROC_TIME from string");
+                    #endif
+                    log_line_parsed->req_proc_time = 0;
+                    log_line_parsed->parsing_errors++;
+                }
+            }
+            else{ // apache time is in microseconds
+                if(unlikely(str2int(&log_line_parsed->req_proc_time, req_proc_time_d, 10) != STR2XX_SUCCESS)) {
+                    #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+                    error("Error while extracting REQ_PROC_TIME from string");
+                    #endif
+                    log_line_parsed->req_proc_time = 0;
+                    log_line_parsed->parsing_errors++;
+                }
+            }
+
+            if(verify){
+                if(unlikely(log_line_parsed->req_proc_time < 0)){
+                    #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+                    error("REQ_PROC_TIME is invalid (<0)");
+                    #endif
+                    log_line_parsed->req_proc_time = 0;
+                    log_line_parsed->parsing_errors++;
+                }
+            }
+            #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+            debug(D_LOGS_MANAG, "Extracted REQ_PROC_TIME:%d", log_line_parsed->req_proc_time);
+            #endif
+
+            goto next_item;
+        }
+
+        if(fields_format[i] == RESP_CODE){
+            #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+            debug(D_LOGS_MANAG, "Item %d (type: RESP_CODE):%.*s\n", i, (int)field_size, field);
+            #endif
+
+            if(unlikely(field[0] ==  '-' && field_size == 1)){
+                log_line_parsed->resp_code = 0;
+                log_line_parsed->parsing_errors++;
+                goto next_item;
+            }
+
+            char resp_code_d[REQ_RESP_CODE_MAX_LEN];
+            snprintf( resp_code_d, 
+                      field_size > (REQ_RESP_CODE_MAX_LEN - 1) ? REQ_RESP_CODE_MAX_LEN : field_size + 1, 
```

Comment:
limits..

Reply:
I am calling attention for limits, because we have a pattern to pass `SIZE-1` to `snprintfz` families.